### DISCUSSION TITLE
How do you use an Enum in your model?

### DISCUSSION THREAD

**Discussion by NateWritesCode** [prisma/nexus-prisma#87] [Q&A] [answered]
Hello, I was wondering how to incorporate an Enum into a model? Would it be best to be explicit using the type of data contained in the Enum (i.e. String)?

 ```t.nonNull.string(ProfessionalCreditSources.name);```
 
 Or is there a way to make it work using the t.field function?
 
 ```t.field(ProfessionalCreditSources.name);```
 
 Thanks for the awesome library!

**Comment by NateWritesCode** [comment, marked as answer]
Figured it out...

First, create an enumType as part of your model

```export const ChatStatusEnumType = enumType(ChatStatus);```

Second, can use t.field like so...

```t.field(Saver.chatStatus.name, { type: ChatStatus.name });```

**Reply by Frosty21** [reply]
don't you mean for the second `{ type: ChatStatus.name });` should be `{ type: ChatStatusEnumType.name });` or just as `enumType(ChatStatus)`  ?

` t.field(Saver.chatStatus.name, { type: ChatStatusEnumType.name }); `

I'm a little confused here...
just comparing it to [prisma-nexus project enums](https://nexus.prisma.io/features#project-enums)

**Reply by NateWritesCode** [reply]
My example has been working for me. ChatStatus comes from:

```
import { ChatStatus } from "nexus-prisma";
```
